botorch/optim/optimize.py

```diff
@@ -305,14 +305,13 @@ def _optimize_batch_candidates() -> Tuple[Tensor, Tensor, List[Warning]]:
 
         # only add parameter constraints to gen_kwargs if they are specified
         # to avoid unnecessary warnings in _filter_kwargs
-        if opt_inputs.inequality_constraints is not None:
-            gen_kwargs["inequality_constraints"] = opt_inputs.inequality_constraints
-        if opt_inputs.equality_constraints is not None:
-            gen_kwargs["equality_constraints"] = opt_inputs.equality_constraints
-        if opt_inputs.nonlinear_inequality_constraints is not None:
-            gen_kwargs[
-                "nonlinear_inequality_constraints"
-            ] = opt_inputs.nonlinear_inequality_constraints
+        for constraint_name in [
+            "inequality_constraints",
+            "equality_constraints",
+            "nonlinear_inequality_constraints",
+        ]:
+            if getattr(opt_inputs, constraint_name) is not None:
+                gen_kwargs[constraint_name] = getattr(opt_inputs, constraint_name)
```

Comment:
This is probably a bad idea since it doesn't work for Python < 3.8, but...

```suggestion
            if  (constraint := getattr(opt_inputs, constraint_name)) is not None:
                gen_kwargs[constraint_name] = constraint
```

Reply:
Nice! The main botorch readme lists Python >= 3.8 as a requirement, should be fine :)